nixos/modules/programs/direnv.nix

```diff
@@ -0,0 +1,111 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}: let
+  cfg = config.programs.direnv;
+in {
+  options.programs.direnv = {
+
+    enable = lib.mkEnableOption (lib.mdDoc ''
+      Whether to enable direnv integration. Takes care of both installation and
+      setting up the sourcing of the shell. Additionally enables nix-direnv
+      integration. Note that you need to logout and login for this change to apply.
+    '');
+
+    direnvrcExtra = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = lib.literalExpression ''
+        export FOO="foo"
+        echo "loaded direnv!"
+      '';
+      description = lib.mdDoc ''
+        Extra lines to append to the sourced direnvrc
+      '';
+    };
+
+    silent = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = lib.mdDoc ''
+        Whether to hide direnv logging
+      '';
+    };
+
+    persistDerivations = lib.mkOption {
```

Comment:
`mkEnableOption` can also be used here.

Reply:
How do set `mkEnableOption` to default to false?

Reply:
Yes thank you